app/controllers/admin/creator_settings_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Admin
+  class CreatorSettingsController < Admin::ApplicationController
+    before_action :extra_authorization, only: [:create]
+
+    ALLOWED_PARAMS = %i[community_name logo_svg primary_brand_color_hex invite_only_mode public].freeze
+
+    def new; end
+
+    def create
+      current_user.saw_onboarding = true
+      current_user.save!
+      ::Settings::Community.community_name = settings_params[:community_name]
+      ::Settings::General.logo_svg = settings_params[:logo_svg]
+      ::Settings::UserExperience.primary_brand_color_hex = settings_params[:primary_brand_color_hex]
+      ::Settings::Authentication.invite_only_mode = settings_params[:invite_only]
+      ::Settings::UserExperience.public = settings_params[:public]
+    rescue StandardError => e
+      flash[:error] = e.message
```

Comment:
What types of errors are we expecting here? Also, should the changes above be performed in a transaction? In other words, if an error happens, are we ok with partial updates or is this an all-or-nothing scenario?

Reply:
We are expecting validation errors, such as the brand color not being dark enough for accessibility reasons, for the settings here. I think that a transaction would be beneficial here, as it is annoying when the form clears out all previous data upon erroring out, @citizen428!